src/main/java/com/qa/koel/HomePage.java

```diff
@@ -3,34 +3,50 @@
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 
 import java.util.List;
 
 public class HomePage extends BasePage{
-    static final By PLAY_NEXT_SONG_BUTTON = By.cssSelector("i[data-testid='play-next-btn']");
-    static final By PLAY_BUTTON = By.cssSelector("span.play");
-    static final By PAUSE_BUTTON = By.cssSelector("span[data-testid='pause-btn']");
-    static final By PROGRESS_BAR= By.cssSelector("div.plyr__progress");
-    static final By PLAYLIST = By.cssSelector("li[class='playlist playlist']");
+
+    @FindBy(css = "i[data-testid='play-next-btn']")
+    WebElement playNextSongButton;
+    @FindBy(css = "span.play")
+    WebElement playButton;
+    @FindBy(css = "span[data-testid='pause-btn']")
+    WebElement pauseButton;
+    @FindBy(css = "div.plyr__progress")
+    WebElement progressBar;
+    @FindBy (css = "li[class='playlist playlist']")
+    WebElement playlist;
+
+   // static final By PLAY_NEXT_SONG_BUTTON = By.cssSelector("i[data-testid='play-next-btn']");
+   // static final By PLAY_BUTTON = By.cssSelector("span.play");
+   //static final By PAUSE_BUTTON = By.cssSelector("span[data-testid='pause-btn']");
+   //static final By PROGRESS_BAR= By.cssSelector("div.plyr__progress");
+   // static final By PLAYLIST = By.cssSelector("li[class='playlist playlist']");
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.

Reply:
I left this code not because I don't understand it, but because I think that in some cases POM works better than pagefactory

Reply:
> I left this code not because I don't understand it, but because I think that in some cases POM works better than pagefactory

Hi

Difference between POM and PageFactory https://www.geeksforgeeks.org/difference-between-pom-and-pagefactory/

Naming convection of methods is user-friendly in the Page Object Model. The Code in POM is easy to understand, reusable and maintainable. In POM, if there is any change in the web element then, it is enough to make the changes in its respective class, rather than editing all the classes.

Pagefactory just like the usual POM is a wonderful concept to apply. However, we need to know where the usual POM is feasible and where Pagefactory suits well. In the static applications (where both XPath and elements are static), Pagefactory can be liberally implemented with added benefits of better performance too.

**The choice between POM and PageFactory depends on the specific requirements and preferences of the project**

Reply:
Thanks for this information